src/bson.ts

```diff
@@ -69,3 +69,32 @@ export function pluckBSONSerializeOptions(options: BSONSerializeOptions): BSONSe
     raw
   };
 }
+
+/**
+ * Merge the given BSONSerializeOptions, preferring options over parentOptions, and optionally
+ * substituting defaults for values not set.
+ *
+ * @internal
+ *
+ * @param options - The primary options to use
+ * @param parentOptions - The secondary options to use
+ * @param includeDefaultOptions - Whether to include default options for any values not set
+ */
+export function inheritBSONOptions(
+  options?: BSONSerializeOptions,
+  parentOptions?: BSONSerializeOptions,
+  includeDefaultOptions?: boolean
```

Comment:
I think we should avoid [magic booleans](https://en.wikipedia.org/wiki/Magic_number_(programming)). Are there cases where we _don't_ want to set defaults? Ultimately we do apply a [set of defaults](https://github.com/mongodb/node-mongodb-native/blob/master/src/cmap/connection.ts#L382-L387) before serializing to the wire protocol, so I'm wondering if we can always apply the defaults.